### PR TITLE
Fix #588 Make OAuth related terms consistent (installPath, redirectUriPath)

### LINKED ISSUE #588: Make OAuth related terms consistent (installPath, redirectUriPath)
This project is the first SDK project that started supporting Slack OAuth flow. The default naming of the URLs are `startPath` and `callbackPath`. A few months later, Bolt for JS decided to go with a bit different names as below:

* `installPath` - default value is `/slack/install`
* `redirectUriPath` - default value is `/slack/oauth_redirect`

See https://slack.dev/bolt-js/concepts#authenticating-oauth for the sources.

Now that Bolt for Python uses the same naming (`install_path` and `redirect_uri_path`), this project is the only one that uses a bit different names.

To fill the gaps, I will introduce the same names as the recommended way since v1.3. **It won't bring any breaking changes. Existing apps can safely use the existing names in v1.x series. We will be removing the legacy names in v2.**

Apart from the two names, I am not planning to apply any other changes to imitate other Bolt frameworks. Other slight differences should be acceptable as they do not represent any unique concepts or terms.

### Category (place an `x` in each of the `[ ]`)

* [x] **bolt** (Bolt for Java)
* [x] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

### Requirements

Please make sure if this topic is specific to this SDK. For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### PR DESCRIPTION
**Pull request by seratch**
This pull request fixes #588. Refer to the issue for details.

### Category (place an `x` in each of the `[ ]`)

* [x] **bolt** (Bolt for Java)
* [x] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

File: bolt/src/main/java/com/slack/api/bolt/AppConfig.java [hunk 1]
```diff
@@ -84,11 +84,36 @@ public boolean isDistributedApp() {
...
     }
 
+    // --------------------------
+
+    @Builder.Default
+    private boolean oAuthInstallPathEnabled = false;
+
+    @Deprecated // will be removed in v2.0 - use oAuthInstallPathEnabled
     @Builder.Default
     private boolean oAuthStartEnabled = false;
```
Comment (seratch) [inline bolt/src/main/java/com/slack/api/bolt/AppConfig.java:94]: I intentionally kept this type of deprecated fields not to remove builder methods.

File: bolt/src/main/java/com/slack/api/bolt/AppConfig.java [hunk 2]
```diff
@@ -84,11 +84,36 @@ public boolean isDistributedApp() {
...
+
+    @Deprecated
+    public boolean isOAuthStartEnabled() {
```
Comment (seratch) [inline bolt/src/main/java/com/slack/api/bolt/AppConfig.java:97]: Explicitly defining getter/setter methods enables us to overwrite the auto-generated methods by Lombok.

File: bolt/src/main/java/com/slack/api/bolt/AppConfig.java [hunk 3]
```diff
@@ -141,30 +156,121 @@ public void setOauthCallbackPath(String oauthCallbackPath) {
...
     }
 
+    // --------------------------
+
+    @Deprecated // will be removed in v2.0 - use getOauthRedirectUriRequestURI instead
     public String getOauthCallbackRequestURI() {
+        return getOauthRedirectUriRequestURI();
+    }
+
+    public String getOauthRedirectUriRequestURI() {
```
Comment (seratch) [inline bolt/src/main/java/com/slack/api/bolt/AppConfig.java:181]: I know this name is a bit weird but, as `RequestURI` is a common term to describe the path of HTTP requests in Java, I kept choosing this suffix here.

File: bolt/src/test/java/test_locally/AppConfigTest.java [hunk 4]
```diff
@@ -11,26 +11,57 @@
     @Test
     public void constructor() {
         AppConfig config = new AppConfig();
+        assertNotNull(config.getOauthInstallPath());
+        assertNotNull(config.getOauthRedirectUriPath());
```
Comment (seratch) [inline bolt/src/test/java/test_locally/AppConfigTest.java:15]: new recommended methods

File: bolt/src/test/java/test_locally/AppConfigTest.java [hunk 5]
```diff
@@ -11,26 +11,57 @@
...
+        assertNotNull(config.getOauthStartPath());
         assertNotNull(config.getOauthCallbackPath());
```
Comment (seratch) [inline bolt/src/test/java/test_locally/AppConfigTest.java:17]: deprecated ones

**Review by seratch — COMMENTED** [review]
a few comments for reviewers

**Review by stevengill — APPROVED** [review]

**Pull request merged**